### PR TITLE
login with facebook

### PR DESCRIPTION
**Pull request by thienbnx-0026**
Login with facebook

### CODE REVIEW

**Comment by thienbnx-0026** [comment]
Screenshot from 2019-08-06 09-08-01

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -30,5 +30,6 @@
 
     <%= render "devise/shared/links" %>
     <p>New user? <%= link_to t("sign_up_now"), new_user_registration_path %></p>
+    <%= link_to "Sign in with facebook", "/auth/facebook", id: "sign_in" %>
```
Comment (quanhd-0035) [inline app/views/devise/sessions/new.html.erb]: i18n

**Comment by thienbnx-0026** [comment]
ready

File: app/models/user.rb [hunk 2]
```diff
@@ -77,6 +77,20 @@ def count_getted_like
...
+    user = User.find_or_create_by(provider: auth.provider, uid: auth.uid)
+
+    user.email = if auth.info.email
+                   auth.info.email
+                 else
+                   auth.uid + "@gmail.com"
+                 end
+    user.name = auth.info.name
+    user.encrypted_password = Devise.friendly_token[0, 20]
+    user.save
```
Comment (phucmh-1356) [inline app/models/user.rb]: nếu user save fail thì sao em

Comment (thienbnx-0026) [reply app/models/user.rb]: Em co dua ra thong bao loi roi a. Xu ly trong ham login_facebook trong session controller

**Comment by thienbnx-0026** [comment]
ready

**Pull request merged**